TFC/parse_workspaces.tf

```diff
@@ -1,7 +1,11 @@
 locals {
   workspaces_from_response = jsondecode(data.external.workspaces.result.workspaces)
   filtered_workspaces_by_name = [
-    for workspace in local.workspaces_from_response : workspace if contains(var.tfc_workspace_names, workspace["attributes"]["name"])
+    for workspace in local.workspaces_from_response : workspace
+    if anytrue([
+      for pattern in var.tfc_workspace_names :
+        can(regex("^${replace(pattern, "*", ".*")}$", workspace["attributes"]["name"]))
```

Comment:
why `can`?
As I understand `can` returns `true` if there is no error and in this case there won't be so it'll be true.

Is this correct?

Reply:
in Terraform, regex returns a list of matches, and if there is no match, it errors. So, to safely check if a string matches a pattern, we use can(regex(...)) to avoid errors

If the workspace name matches the pattern, regex succeeds and can() returns true.
If it doesn't match, regex errors and can() returns false.